src/web/app/pages-instructor/instructor-search-page/instructor-search-bar/instructor-search-bar.component.ts

```diff
@@ -37,6 +37,9 @@ export class InstructorSearchBarComponent implements OnInit {
    * send the search data to parent for processing
    */
   search(): void {
+    if (!(this.searchParams.isSearchForComments ||
+        this.searchParams.isSearchForStudents) ||
+        this.searchParams.searchKey === '') { return; }
```

Comment:
Probably better to let the parent components handle this check? The child component just emits the event and not have to bother with validation